frontend/catalyst/pennylane_extensions.py

```diff
@@ -395,15 +400,27 @@ def __call__(self, *args, **kwargs):
 
         if EvaluationContext.is_tracing():
             fn = _ensure_differentiable(self.fn)
+            argnum_is_int = (
+                isinstance(self.grad_params.argnum, int) or self.grad_params.argnum is None
+            )
```

Comment:
Looks like this can be computed in the `unflatten_derivatives` function since it's not used here.

Reply:
it cannot in fact, because we are putting the argnum in a list in check_grad_params

Reply:
I have change the system to have different argnums, see the comment where I have tagged you